contracts/src/v0.8/shared/call/CallWithExactGas.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/// @dev Handles the edge case where we want to pass a specific amount of gas,
+/// @dev but EIP-150 sends all but 1/64 of the remaining gas instead so the user gets
+/// @dev less gas than they paid for. The other 2 parts of EIP-150 do not apply since
+/// @dev a) we hard code value=0 and b) we ensure code already exists.
+/// @dev If we revert instead, then that will never happen.
+/// @dev Separately we capture the return data up to a maximum size to avoid return bombs,
+/// @dev borrowed from https://github.com/nomad-xyz/ExcessivelySafeCall/blob/main/src/ExcessivelySafeCall.sol.
+library CallWithExactGas {
```

Comment:
Note that there is another extra variant not captured here, might be worth including?

https://github.com/smartcontractkit/libocr/blob/ffa38fe11ef8e9571e20f4ce4166e7010cfbe885/contract2/dev/OCR2Aggregator.sol#L429

Reply:
I'll add that one as well!